fairscale/nn/pipe/async_pipeline.py

```diff
@@ -4,18 +4,54 @@
 # LICENSE file in the root directory of this source tree.
 
 import logging
+import os
 from threading import Event
-from typing import List, Optional
+from typing import Dict, List, Optional, Union
 
 import torch
 
-from .async_schedule import AsyncEventLoop
+from .async_schedule import AsyncEventLoop, ModuleWrapper
+from .messages import MakeTransport
 from .microbatch import Batch
-from .multiprocess_pipeline import MultiProcessPipeline
+from .skip.layout import SkipLayout
 from .skip.tracker import SkipTrackerThroughPotals
 
 
-class AsyncPipeline(MultiProcessPipeline):
+class AsyncPipeline:
```

Comment:
Why do we not need to subclass this anymore?

Reply:
I want to break the cyclic dependency between multiprocess_pipeline and async_schedule. The class hierarchy is a bit of a mess currently. I also don't like that the run function of *Pipeline takes a different argument type in the two classes.

Removing the inheritance should enables implication of MultiProcess pipe in order to understand what is really going on. I may add the inheritance back later, or add a common interface base-class, if it makes sense but right now it just gets in the way.